# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/maryam12arshad17/flyrank-internship-ml/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane (or Freestyle) and Why

I chose the **Refresh / Content Opportunity Scoring** lane because it focuses on identifying web pages that may need a content refresh. Large websites often contain hundreds or thousands of pages, making manual review difficult and time-consuming. Machine learning can analyze multiple search performance signals and rank pages that are more likely to benefit from updates. This helps content and SEO teams prioritize their work more efficiently. I selected this lane because it aligns with the starter dataset and builds on the work completed in the Week 1 notebooks.

In [20]:
print("Selected Lane: Refresh / Content Opportunity Scoring")


Selected Lane: Refresh / Content Opportunity Scoring


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*


### Research Question
Which web pages should be prioritized for content refresh based on the available search performance data?

### Unit of Analysis
Each row in the dataset represents one web page.

### Output
A ranked list or score indicating which pages should be reviewed first for content refresh.

### Decision
Determine which pages should be refreshed before others.

### Action
The content or SEO team can review, update, improve, or monitor the recommended pages.

### Cost of a Wrong Recommendation
If the recommendation is incorrect, important pages may be ignored while time and effort are spent updating pages that do not require immediate attention. This can reduce productivity and delay performance improvements.

### Why Machine Learning?
Large websites contain many pages and multiple performance signals, making manual analysis difficult. Machine learning can combine these signals to identify pages that are more likely to need a content refresh and help teams make faster and more consistent decisions.

In [21]:
print("Research question, decision, and action defined.")


Research question, decision, and action defined.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [22]:
import pandas as pd

# Use the same CSV path from Notebook 1
df = pd.read_csv("content_refresh_anonymized.csv")

print("Total Pages:", len(df))
print("Average CTR:", round(df["ctr"].mean(),2))
print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())


Total Pages: 30000
Average CTR: 0.51

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


# Quick Look at the Data (2–3 Real Numbers)

The starter dataset contains **30,000 web pages**, providing a large amount of data for analysis. The **average CTR is 0.51**, showing that click-through rates vary across pages. The trend distribution shows that **16,262 pages are declining**, while **4,388 pages are improving** and **5,962 pages are stable**. These numbers suggest that many pages may require review or content updates, making the **Refresh / Content Opportunity Scoring** lane a meaningful project for the next seven weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Words

This project provides decision support by identifying pages that may benefit from a content refresh based on patterns observed in the available data. The results should be interpreted as recommendations rather than facts. The model cannot prove that refreshing a page will improve rankings or traffic, and it cannot predict future search rankings or Google's internal ranking algorithm. Final decisions should always involve human review and business judgment.

In [23]:
print("This project provides decision support, not guaranteed predictions.")


This project provides decision support, not guaranteed predictions.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.